### PR TITLE
AP_Periph: add support for RawAirData.Pitot_temperature

### PR DESCRIPTION
**Pull request by magicrub**
AP_Periph: add support for RawAirData.Pitot_temperature in outbound packets

As of now, Ardupilot does not yet have any inbound use for this yet - it's for reporting only. There is a coming PR for a thermostat-based heated pitot tube to take advantage of this.

### CODE REVIEW

File: libraries/AP_TemperatureSensor/AP_TemperatureSensor_Backend.cpp [hunk 1]
```diff
@@ -89,7 +89,6 @@ void AP_TemperatureSensor_Backend::update_external_libraries(const float tempera
             break;
 #endif
 
-        case AP_TemperatureSensor_Params::Source::None:
```
Comment (peterbarker) [inline libraries/AP_TemperatureSensor/AP_TemperatureSensor_Backend.cpp:92]: Can we get rid of the default: case instead here?

Comment (magicrub) [reply libraries/AP_TemperatureSensor/AP_TemperatureSensor_Backend.cpp:92]: LOL I went back and forth on this. Yes, I'll list every enum to make the class happy.

Comment (magicrub) [reply libraries/AP_TemperatureSensor/AP_TemperatureSensor_Backend.cpp:92]: done

**Comment by magicrub** [comment]
A nice future TODO is to add the concept of a pitot tube temperature to the airspeed driver.

**Comment by magicrub** [comment]
@peterbarker bump

File: libraries/AP_TemperatureSensor/AP_TemperatureSensor_Backend.cpp [hunk 2]
```diff
@@ -63,34 +63,41 @@ void AP_TemperatureSensor_Backend::set_temperature(const float temperature)
...
 void AP_TemperatureSensor_Backend::update_external_libraries(const float temperature)
 {
-#if HAL_WITH_ESC_TELEM
-    AP_ESC_Telem_Backend::TelemetryData t;
-#endif
-
     switch ((AP_TemperatureSensor_Params::Source)_params.source.get()) {
-#if HAL_WITH_ESC_TELEM
         case AP_TemperatureSensor_Params::Source::ESC:
+#if HAL_WITH_ESC_TELEM
```
Comment (peterbarker) [inline libraries/AP_TemperatureSensor/AP_TemperatureSensor_Backend.cpp:68]: This is not the right way to remove the default clause; the enumeration values should be removed when the source is not available.

The problem with changing things like this is that it costs bytes for a feature even when it's not compiled in.  Removing the enumeration entry, on the other hand, has the advantage that there can't be *any* code left in which is gated on that enumeration value.

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: libraries/AP_TemperatureSensor/AP_TemperatureSensor_Backend.cpp (hunk 2 above)
Comment (magicrub) [reply libraries/AP_TemperatureSensor/AP_TemperatureSensor_Backend.cpp]: I've removed the commit per your suggestion on discord

**Comment by magicrub** [comment]
@peterbarker bump

**Review by peterbarker — APPROVED** [review]
Code looks right.

Please state what testing has been done on this.

**Comment by magicrub** [comment]
I have a custom Periph HW with a custom heated pitot tube that has it's own Type-K thermopile temperature sensor touching the metal pitot tube that is the feedback for a thermostat. Knowing the pitot tube temperature for anti-icing is different than knowing the "air temp" from the differential baro which is sampled on the end of along-ish tube inside the aircraft. Now I get some visibility of what that thermostat is doing

**Pull request merged**